lib/commands/version_commands.bash

```diff
@@ -33,15 +33,28 @@ version_command() {
   check_if_plugin_exists "$plugin_name"
 
   declare -a resolved_versions
-  local version
-  for version in "${versions[@]}"; do
-    if [ "$version" = "latest" ]; then
+  local item
+  for item in "${!versions[@]}"; do
+    IFS=':' read -r -a version_info <<<"${versions[$item]}"
+    if [ "${version_info[0]}" = "latest" ] && [ -n "${version_info[1]}" ]; then
+      version=$(asdf latest "$plugin_name" "${version_info[1]}")
+    elif [ "${version_info[0]}" = "latest" ] && [ -z "${version_info[1]}" ]; then
       version=$(asdf latest "$plugin_name")
+    else
+      # if branch handles ref: || path: || normal versions
+      version="${versions[$item]}"
+    fi
+
+    # check_if_version_exists should probably handle if either param is empty string
+    if [ -z "$version" ]; then
+      exit 1
     fi
+
     if ! (check_if_version_exists "$plugin_name" "$version"); then
       version_not_installed_text "$plugin_name" "$version" 1>&2
       exit 1
     fi
```

Comment:
this `check_if_version_exists` handles the `ref:<ref>` & `path:<dir>` filters two calls deeper in the stack. Those I feel should be elevated to this level as `latest:<filter>` is handled here.

Reply:
Not suggestion doing that now, but this codebase is becoming spaghetti as everything gets pushed down into `utils.bash`